spec.md

```diff
@@ -1827,8 +1960,20 @@ operand types are valid and how to convert the operand `x` to a string:
 s       any             as if by str(x)
 r       any             as if by repr(x)
 d       number          signed integer decimal
+o       number          signed octal
+e       number          float exponential format, lowercase (1.230000e+12)
+E       number          float exponential format, uppercase (1.230000E+12)
+f       number          float decimal format                (1230000000000.000000)
+F       number          same as %f
+g       number          compact format, lowercase           (0.0, 1.1, 1200, 1e+45, 1.2e+12) 
+G       number          compact format, lowercase           (0.0, 1.1, 1200, 1e+45, 1.2E+12)
 ```
 
+The compact form `%g` is also used by `str(float)`. Its result
+uses the least precision required to accurately represent the value,
+omits unnecessary trailing zeros in the mantissa, and always contains a period
```

Comment:
Nit: omitting unnecessary trailing zeros means that `"%g" % 5.0 == "5."`, rather than `"5.0"`.

Reply:
Now: "omits unnecessary trailing zeros in the significand"

Reply:
Hair splitting, but that sounds the same as "uses the least precision required", and also doesn't seem to exclude `"5."`. You could just drop this middle clause, and expand the "contains a period" case to say a period with at least one following digit.

Reply:
Changed to:
```                                                                                             
Its result uses the least precision required to unambiguously represent the value,                              
omits unnecessary trailing zeros in the significand,                                                 
and the decimal point itself if the significand has no fraction,                                     
and always contains a decimal point or an exponent and thus                                          
unambiguously denotes a `float`, not an `int`.                                                       
```